packages/api/endpoints/granules.js

```diff
@@ -140,37 +141,50 @@ async function del(req, res) {
   log.info(`granules.del ${granuleId}`);
 
   const granuleModelClient = new models.Granule();
+  const granulePgModel = new GranulePgModel();
+  const filePgModel = new FilePgModel();
 
-  let granule;
+  const knex = await getKnexClient({ env: process.env });
+
+  let dynamoGranule;
+  let pgGranule;
   try {
-    granule = await granuleModelClient.getRecord({ granuleId });
+    dynamoGranule = await granuleModelClient.getRecord({ granuleId });
+    pgGranule = await granulePgModel.get(knex, { granule_id: granuleId });
   } catch (error) {
     if (error instanceof RecordDoesNotExist) {
       return res.boom.notFound(error);
     }
     throw error;
   }
 
-  if (granule.detail) {
-    return res.boom.badRequest(granule);
+  // TODO what is this?
+  if (dynamoGranule.detail) {
+    return res.boom.badRequest(dynamoGranule);
   }
 
-  try {
-    await granuleModelClient.delete(granule);
-  } catch (error) {
-    if (error instanceof DeletePublishedGranule) {
-      return res.boom.badRequest(error.message);
-    }
-    throw error;
-  }
+  await knex.transaction(async (trx) => {
+    // Delete the granule's files from Postgres and S3
+    await filePgModel.deleteGranuleFiles(trx, pgGranule);
+    // Delete the Granule from Postgres
+    await granulePgModel.delete(trx, pgGranule);
+    // Delete the granule from Dynamo
+    await granuleModelClient.delete(dynamoGranule);
+  })
+    .catch((error) => {
+      if (error instanceof DeletePublishedGranule) {
+        return res.boom.badRequest(error.message);
+      }
+      throw error;
+    });
```

Comment:
Yeah, that is definitely more readable. This is actually a little different in the latest refactor though. I'm checking if the granule is published immediately in the endpoint:

https://github.com/nasa/cumulus/pull/2057/files#diff-624123a7dc5bb5513e3769b27fe85821577087609a9376829676b7edbeb3a5aeR167-R169

It's set up this way because we're deleting files first and it wouldn't really make sense for the File's delete method to be responsible for checking the published status of a granule.